src/components/inline-asset-search/inline-asset-search.tsx

```diff
@@ -90,7 +91,7 @@ export const InlineAssetSearch = ({
             >
               <>
                 {options.map((x, i) => (
-                  <option key={i} value={x.value}>
+                  <option key={i} value={JSON.stringify(x)}>
                     {x.label}
                   </option>
                 ))}
```

Comment:
Why is a JSON string being used for the value here?

Reply:
Unfortunately you can't just set `value = {x}`. Option elements want their value to be a string, and the simplest way to select the whole object, rather than just its GUID, was to stringify the whole thing in my opinion

Reply:
Serializing JSON is expensive, especially on a long list. Using the GUID would be better

Reply:
GUID is what was being used (x.value was the GUID) and that was fine for just populating `parentAssetIds` but now we need to also populate `assetLocation.parentAsset` which means we need more information from the selected parent, rather than just the GUID.
If it becomes an issue we can optimize it later down the light, however the JSON that's being serialized is only a very small string in this case. Example below:
![image](https://github.com/LBHackney-IT/mtfh-frontend-property/assets/70756861/9d73aafa-2972-4aa6-b049-dac339d88fbd)

Reply:
The issue is you're passing data through the code by stringifying an object and then parsing it back. This makes the code very hard to read. I certainly didn't make the connection between the option value and getParentAsset in requestFactory.

You may be right that the performance penalty is negligible, but it is still unnecessary. A much simpler solution would be to use the id, and then use that to filter the correct asset. The flow of data would be much more readable